arduino/src/arduino.cpp

```diff
@@ -0,0 +1,444 @@
+/**
+ * @file arduino.cpp
+ * @author yarin
+ * @brief  the arduino<-> pi protocol arduino object
+ * @version 0.1
+ * @date 2020-11-21
+ * @copyright Copyright (c) 2020
+ * 
+ */
+
+#include "arduino.h"
+#include <ctime>
+#define RATE 9600
+#define TIMEOUT 10000
+#define READLENGTH 100
+#define SUCCESS 1
+#define DATALEN 1000
+#define COMMENDHEADER "3"
+#define FAILURE -1
+#define NULLCHAR '\0'
+#define CHECKBUFFSIZE 64
+
+/*!
+    @brief constract the arduino class for the testing
+*/
+Arduino::Arduino()
+{
+    this->portName = "/dev/ttyUSB0";
+    this->buadRate = RATE;
+    this->timeout = TIMEOUT;
+}
+/*!
+ * @brief Destroy the Arduino:: Arduino object 
+ * probably meaningless 
+ * //TODO the meaning of this is to be determined
+
+ */
+Arduino::~Arduino()
+{
+    this->serial.closeDevice();
+    this->isOpen = false;
+}
+/**
+ * @brief Construct a new Arduino:: Arduino object
+ * 
+ * @param fileToWrite 
+ * @param portName 
+ * @param buadRate 
+ * @param timeout 
```

Comment:
i know what the parameters are by the signature.
if you want to write @parm you also need to write an expiation what is each parameter